python/lsst/ts/atspectrograph/atspec_csc.py

```diff
@@ -635,61 +636,97 @@ async def configure(self, config):
 
         self.model.tolerance = config.tolerance
 
-        if len(config.filters) == len(ATSpectrograph.FilterPosition)-1:
-            self.model.filters = dict()
+        # Verify configurations for filter_to_enum_mapping_dict are populated correctly.
+        # create dictionary mapping the name to the filter position
+        if (len(config.filters['filter_name']) == len(ATSpectrograph.FilterPosition) - 1)and \
+                (len(config.filters['filter_central_wavelength']) == len(ATSpectrograph.FilterPosition) - 1) and \
+                (len(config.filters['filter_focus_offset']) == len(ATSpectrograph.FilterPosition) - 1):
+            self.model.filter_to_enum_mapping_dict = dict()
+            self.filter_info = config.filters
+            self.grating_info = config.gratings
+
+            # create relationship between filter name and enumeration since either can be used as inputs
             for i, f in enumerate(ATSpectrograph.FilterPosition):
-                print(self.model.filters)
-                self.model.filters[config.filters[i]] = f
+                self.model.filter_to_enum_mapping_dict[config.filters['filter_name'][i]] = f
+
+                # Why do this? Appears enums can go larger than 3?
```

Comment:
If you do:

```
In [3]: for i, f in enumerate(ATSpectrograph.FilterPosition):
   ...:     print(i, f)
   ...:
0 FilterPosition.FILTER0
1 FilterPosition.FILTER1
2 FilterPosition.FILTER2
3 FilterPosition.FILTER3
4 FilterPosition.INBETWEEN
```

You only want to get the first 3 items and ignore the `INBETWEEN`. 

now, `len(ATSpectrograph.FilterPosition) == 5`,  so, to ignore the last one you need `i < len(ATSpectrograph.FilterPosition)-2`.